### PR TITLE
IEx uses erlang shell rendering for erlang docs

### PR DESCRIPTION
**Pull request by sabiwara**
As suggested [here](https://github.com/elixir-lang/elixir/pull/12802#issuecomment-1637849231), use [shell_docs:render/3](https://www.erlang.org/doc/man/shell_docs.html#render-2) to render Erlang docs in IEx:

Screenshot 2023-07-17 at 21 09 58

### CODE REVIEW

**Comment by josevalim** [comment]
I think there is a bunch of code in IO.ANSI.Docs you can now remove, as it no longer deals with Erlang AST. It may be worth doing a pass on the code to find other areas that can now be removed (perhaps blame when the feature was added and see related areas?).

**Comment by sabiwara** [comment]
Sounds good, will do !

**Comment by sabiwara** [comment]
> I think there is a bunch of code in IO.ANSI.Docs you can now remove, as it no longer deals with Erlang AST. It may be worth doing a pass on the code to find other areas that can now be removed (perhaps blame when the feature was added and see related areas?).

OK, so before we can clean up we'll also need to migrate the module doc and type docs (`t`) to use `:shell_docs`.
I think it makes sense to migrate thoroughly, just wanted to confirm before I move forward.

Screenshot 2023-07-18 at 8 19 03

----

Screenshot 2023-07-18 at 8 17 56

**Comment by josevalim** [comment]
Correct. We can print all Erlang entries with Erlang tools.

**Review by joaothallis — APPROVED** [review]

**Comment by sabiwara** [comment]
@josevalim I think we should be good:

- added support for erlang module doc
- added support for erlang types
- cleaned up the ansi docs erlang support added in https://github.com/elixir-lang/elixir/pull/9843/files

Screenshot 2023-07-19 at 23 07 57

Screenshot 2023-07-19 at 23 08 26

File: lib/iex/lib/iex/introspection.ex [hunk 1]
```diff
@@ -659,68 +666,86 @@ defmodule IEx.Introspection do
   Prints the types for the given module and type documentation.
   """
   def t(module) when is_atom(module) do
-    case Typespec.fetch_types(module) do
-      :error ->
-        no_beam(module)
+    case :code.get_doc(module) do
+      {:ok, {:docs_v1, _, :erlang, _, _, _, _} = erl_docs} ->
+        :shell_docs.render_type(module, erl_docs) |> IO.puts()
```
Comment (sabiwara) [inline lib/iex/lib/iex/introspection.ex:671]: Note: I've tried to refactor these  three `t` clauses but each implementation is slightly different  in a way that makes it quite difficult to abstract, so I ended up just nesting the whole implementation in an extra case.

**Pull request merged**